core/nlp/spacy_tokenizer.py

```diff
@@ -111,57 +139,160 @@ class MultilingualTokenizer:
+            return ["tok2vec", "tagger", "attribute_ruler", "lemmatizer"]
+
+    def _activate_components_to_lemmatize(self, language: AnyStr) -> None:
+        """Set the components of SpaCy Language to lemmatize text:
+        - If SpaCy Language is load as a pre-trained model: enable tok2vec, morphologizer and lemmatizer
+        - Else: Add a Lemmatizer in the available mode 'rule' or 'lookup'
+        (cf https://github.com/explosion/spacy-lookups-data/tree/master/spacy_lookups_data/data )
+        """
+        if self.use_models and language in SPACY_LANGUAGE_MODELS_LEMMATIZATION:
+            # When using a pre-trained model
+            components_to_activate = self._get_components_to_activate_lemmatization(language)
+            if language in self._restore_pipe_components:
+                self._restore_pipe_components[language].restore()
+            self._restore_pipe_components[language] = self.spacy_nlp_dict[language].select_pipes(
+                enable=components_to_activate
+            )
+        elif language in SPACY_LANGUAGE_LOOKUP:
+            # When using spacy lookups
+            self.spacy_nlp_dict[language].add_pipe("lemmatizer", config={"mode": "lookup"})
+            self.spacy_nlp_dict[language].initialize()
+        elif language in SPACY_LANGUAGE_RULES:
+            # When using spacy rules (e.g in case there is no spacy lookup available)
+            self.spacy_nlp_dict[language].add_pipe("lemmatizer", config={"mode": "rule"})
+            self.spacy_nlp_dict[language].initialize()
+        else:
+            # unsupported cases
+            raise ValueError(self._get_error_message_lemmatization(language))
 
     def _create_spacy_tokenizer(self, language: AnyStr) -> Language:
         """Private method to create a custom spaCy tokenizer for a given language
-
         Args:
             language: Language code in ISO 639-1 format, cf. https://spacy.io/usage/models#languages
-
         Returns:
             spaCy Language instance with the tokenizer
-
         Raises:
             TokenizationError: If something went wrong with the tokenizer creation
-
         """
         start = perf_counter()
         logging.info(f"Loading tokenizer for language '{language}'...")
         try:
-            if language == "th":  # PyThaiNLP requires a "data directory" even if nothing needs to be downloaded
+            if (
+                language == "th"
+            ):  # PyThaiNLP requires a "data directory" even if nothing needs to be downloaded
                 os.environ["PYTHAINLP_DATA_DIR"] = mkdtemp()  # dummy temp directory
             if language in SPACY_LANGUAGE_MODELS and self.use_models:
                 nlp = spacy.load(SPACY_LANGUAGE_MODELS[language])
+            # Since v3, spaCy uses char tokenization by default for Chinese, not jieba anymore
+            # See https://github.com/explosion/spaCy/blob/e1f88de729f113f068958c824cf01026363bb110/spacy/lang/zh/__init__.py
+            elif language == "zh":
```

Comment:
Since you are using an elif, what is happening ` if language in SPACY_LANGUAGE_MODELS and self.use_models`?
Does it still use jieba?

Reply:
I think in that case it will use a statistical model to do the splitting without jieba - It is not recommended to use jieba together with the model, see https://github.com/explosion/spaCy/discussions/8577#discussioncomment-955726

Reply:
Sounds good, thanks for checking. I would mention what you explained explicitly in a comment, for future readers.

Reply:
Good idea, done!